src/picterra/__main__.py

```diff
@@ -89,21 +92,75 @@ def parse_args(args):
         "--output-type", help="Output type of the detector",
         type=str, choices=['polygon', 'bbox'], default='polygon')
     create_detector_parser.add_argument(
-        "--training-steps", metavar='',
-        help="Number of steps while training the detector: an integer in the range 500..40000",
-        type=int, default=500)
+        "--training-steps", type=int, default=500,
+        help="Number of steps while training the detector: an integer in the range 500..40000")
     create_detector_parser.add_argument(
         "-r", "--raster", help="ID(s) of the raster (s) to associate with the detector",
         type=str, required=False, nargs='+', default=[])
-    # create raster
+    # Create raster
     create_raster_parser = create_subparsers.add_parser(
         'raster', help="Creates a raster, optionally adding it to a detector")
-    create_raster_parser.add_argument("path", help="Path to the raster file", type=str)
-    create_raster_parser.add_argument("--name", help="Name of the raster", type=str, required=False)
-    create_raster_parser.add_argument(
+    create_raster_subparsers = create_raster_parser.add_subparsers(dest='raster')
+    # create raster (from file)
+    create_raster_from_file_parser = create_raster_subparsers.add_parser(
+        'file', help="Uploads raster from a local file")
+    create_raster_from_file_parser.add_argument("path", help="Path to the raster file", type=str)
+    create_raster_from_file_parser.add_argument(
+        "--name", help="Name to give to the raster", type=str, required=False)
+    create_raster_from_file_parser.add_argument(
+        "--folder", help="Id of the folder/project to which the raster will be uploaded",
+        type=str, required=False)
+    create_raster_from_file_parser.add_argument(
+        "-d", "--detector", help="ID(s) of the detector(s) to which we'll associate the raster",
+        type=str, required=False, nargs='+', default=[])
+    # create raster (from remote imagery server)
+    create_raster_from_remote_parser = create_raster_subparsers.add_parser(
+        'remote', help="Uploads raster from a remote imagery server")
+    create_raster_from_remote_parser.add_argument(
+        "type", help="Type of the imagery server", type=str, choices=['wms', 'xyz'])
+    create_raster_from_remote_parser.add_argument("url", help="URL of the imagery server", type=str)
+    create_raster_from_remote_parser.add_argument(
+        "resolution", help="Spatial resolution (GSD) of the raster to upload, in meters",
+        type=float)
+
+    def create_remote_raster_type(string):
```

Comment:
Perhaps rename `string` to something more explicit, like `coords_string`?

Reply:
Checked